powersimdata/design/generation/curtailment.py

```diff
@@ -0,0 +1,124 @@
+import pandas as pd
+
+from powersimdata.scenario.scenario import Scenario
+
+default_pmin_dict = {
+    "coal": None,
+    "dfo": 0,
+    "geothermal": 0.95,
+    "hydro": None,
+    "ng": 0,
+    "nuclear": 0.95,
+    "other": 0,
+    "solar": 0,
+    "wind": 0,
+    "wind_offshore": 0,
+}
+profile_methods = {
+    "hydro": "get_hydro",
+    "solar": "get_solar",
+    "wind": "get_wind",
+    "wind_offshore": "get_wind",
+}
+
+
+def temporal_curtailment(
+    scenario, pmin_by_type=None, pmin_by_id=None, curtailable={"solar", "wind"}
+):
+    """Calculate the minimum share of potential renewable energy that will be curtailed
+        due to supply/demand mismatch, assuming no storage is present.
+
+    :param powersimdata.scenario.scenario.Scenario scenario: scenario instance.
+    :param dict/pandas.Series pmin_by_type: Mapping of types to Pmin assumptions. Values
+        between 0 and 1 (inclusive) are treated as shares of Pmax, and None values
+        either maintain given Pmin values (for dispatchable resources) or track profiles
+        (for profile resources, e.g. hydro).
+    :param dict/pandas.Series pmin_by_id: Mapping of IDs to Pmin assumptions, as an
+        override to the default behavior for that plant type. Values between 0 and 1
+        (inclusive) are treated as shares of Pmax, and None values either maintain given
+        Pmin values (for dispatchable resources) or track profiles
+        (for profile resources, e.g. hydro).
+    :param iterable curtailable: resource types which can be curtailed.
+    :return: (*float*) -- share of curtailable resources that will be curtailed.
+    :raises TypeError: if inputs do not mach specified type.
+    :raises ValueError: if any entries in curtailable or keys in pmin_by_type are not
+        types in the grid, any keys in pmin_by_id are not plant IDs in the Grid,
+        or any values in pmin_by_type/pmin_by_id are not in the range [0, 1] or None.
+    """
+    if not isinstance(scenario, Scenario):
+        raise TypeError("scenario must be a Scenario")
+    if pmin_by_type is None:
+        pmin_by_type = {}
+    if pmin_by_id is None:
+        pmin_by_id = {}
+    check_dicts = {"pmin_by_id": pmin_by_id, "pmin_by_type": pmin_by_type}
+    for name, d in check_dicts.items():
+        if not isinstance(d, (dict, pd.Series)):
+            raise TypeError(f"{name} must be a dict or pandas Series")
+        if not all([v is None or 0 <= v <= 1 for v in d.values()]):
```

Comment:
It seems that `d` can either be a `dict` or a `pd.Series`. So if it is a `pd.Series`, `d.values()` won't work. No?

Reply:
You're right, I had originally thought that we can be flexible on inputs because a Series has an equivalent `items()` method, but it does not have an equivalent `values()` method.

Reply:
I guess we will test all that when you have time to do write the test module

Reply:
I added a small change that should fix this issue.